File: languages/go/entity/key/ecdsa_key_pair.go
package key

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

func NewEcdsaKeysFromProto(k *proto.GenerateLocalKeyResponse) KeyPair {
	if k == nil {
		return KeyPair{}
	}
	return KeyPair{
		PublicKey:  k.LocalKey.Key,
		PrivateKey: *k.LocalKey.PrivateKey,
	}
}

File: languages/js/src/entity/key/ecdsa_key_pair.ts
import * as keyProto from "../../bridge/proto/bloock_keys";
import { KeyPair } from "./key_pair";

/**
 * Represents an ecdsa key pair, with private and public key.
 */
export class EcdsaKeyPair extends KeyPair {
  static fromProto(k: keyProto.GenerateLocalKeyResponse): KeyPair {
    return new KeyPair(k.localKey!.key, k.localKey!.privateKey!);
  }
}
